Make pytorch clang-tidy clean

1ntEgr8 opened the pull request:
This PR suppresses clang-tidy warnings in the codebase (for now) so that we can re-enable clang-tidy checks on master.

I ran this script to add the `NOLINTNEXTLINE` comments (on a devserver):
```bash
python3 setup.py develop

# Uses same script that's run on CI and adds the -j (parallel), -s (add comments), -k (continue if diagnostic errors are found) options
python3 tools/clang_tidy.py \
  -j \
  -s \
  -k \
  -v \
  --paths torch/csrc/ \
  -g"-torch/csrc/jit/passes/onnx/helper.cpp" \
  -g"-torch/csrc/jit/passes/onnx/shape_type_inference.cpp" \
  -g"-torch/csrc/jit/serialization/onnx.cpp" \
  -g"-torch/csrc/jit/serialization/export.cpp" \
  -g"-torch/csrc/jit/serialization/import.cpp" \
  -g"-torch/csrc/jit/serialization/import_legacy.cpp" \
  -g"-torch/csrc/onnx/init.cpp" \
  -g"-torch/csrc/cuda/nccl.*" \
  -g"-torch/csrc/cuda/python_nccl.cpp" \
  -g"-torch/csrc/autograd/FunctionsManual.cpp" \
  -g"-torch/csrc/generic/*.cpp" \
  -g"-torch/csrc/jit/codegen/cuda/runtime/*" \
  -g"-torch/csrc/deploy/interpreter/interpreter.cpp" \
  -g"-torch/csrc/deploy/interpreter/interpreter.h" \
  -g"-torch/csrc/deploy/interpreter/interpreter_impl.h" \
  -g"-torch/csrc/deploy/interpreter/test_main.cpp"
```

Test plan:
Verified changes by re-running the script (without the `-s` option) and seeing no warnings/errors.

janeyx99 commented:
From the lint job, it looks like there are still some clang-tidy errors that should be fixed, like in `torch/csrc/deploy/interpreter/interpreter_impl.cpp`

malfet commented on torch/csrc/CudaIPCTypes.cpp:
```diff
@@ -32,6 +32,7 @@ struct CudaIPCGlobalEntities {
   std::shared_ptr<CudaIPCRefCountersFile> next_available_ref_counters_file_;
   CudaIPCSentDataLimbo CudaIPCSentDataLimbo_;
   // NOLINTNEXTLINE(cppcoreguidelines-pro-type-member-init)
+  // NOLINTNEXTLINE(modernize-use-equals-default)
```
```suggestion
  // NOLINTNEXTLINE(cppcoreguidelines-pro-type-member-init,modernize-use-equals-default)
```
You need to modify annotation script to merge two NOLINTNEXTLINE statements together

driazati commented on torch/csrc/autograd/forward_grad.h:
```diff
@@ -84,17 +84,18 @@ struct ForwardGrad;
-    void erase(const std::shared_ptr<ForwardGrad>& grad) {
-        std::lock_guard<std::mutex> lock(mutex_);
-        grads_.erase(grad);
+  // NOLINTNEXTLINE(cppcoreguidelines-pro-type-member-init)
```
the formatting seems to have been messed up here (not that it really matters, we can auto clang format this), but was that the result of the script inserting the `NOLINTNEXTLINE`s?

1ntEgr8 replied:
Yes, the script inserted those. I'll auto clang format this

driazati reviewed (approved):
lgtm, some minor formatting weirdness but I don't want this to get stuck rebasing over and over

guotuofeng committed:
escape backward slash in stack trace in Windows to slash (#60842)

Summary:
Fixes https://github.com/pytorch/pytorch/issues/60187

Pull Request resolved: https://github.com/pytorch/pytorch/pull/60842

Reviewed By: gdankel

Differential Revision: D29498498

Pulled By: malfet

fbshipit-source-id: 78e1b25a2e6bdfd3ba0c988d023c7a7f79a22cf4

malfet commented:
Hmm, this PR introduced a lot of false-positive clang-tidy warnings, for example:
```
  // NOLINTNEXTLINE(cppcoreguidelines-init-variables)
  std::unique_ptr<CudaIPCSentData> sent_data(
      static_cast<CudaIPCSentData*>(ptr));
```
